### PR TITLE
chore: add custom metrics to sgController

### PR DESCRIPTION
**Pull request by Rdpaula**
This issue aims to add a custom metric to track consecutive reconciliation errors with labels of controller name, environment, and object name so that we can group the errors in a more granular way and filter the environments.

### CODE REVIEW

File: internal/controller/ec2.aws/securitygroup_controller.go [hunk 1]
```diff
@@ -181,7 +188,7 @@ func (r *SecurityGroupReconciliation) retrieveKCPNetworkInfo(ctx context.Context
 		Name:      name,
 		Namespace: namespace,
 	}
-	if err := r.Client.Get(ctx, key, kcp); err != nil {
+	if err := r.Get(ctx, key, kcp); err != nil {
```
Comment (csmartins) [inline internal/controller/ec2.aws/securitygroup_controller.go:191]: was this wrong before?

File: internal/controller/ec2.aws/securitygroup_controller.go [hunk 2]
```diff
@@ -967,3 +974,45 @@ func (r *SecurityGroupReconciliation) addKCPFinalizer(ctx context.Context, kcp k
 		}
 	}
 }
+
+func (r *SecurityGroupReconciliation) getEnvironment(ctx context.Context) string {
```
Comment (csmartins) [inline internal/controller/ec2.aws/securitygroup_controller.go:978]: Do we need to go through all the references to get the environment? Isn't it enough to get the environment of the first?

File: pkg/kops/kops.go [hunk 3]
```diff
@@ -24,12 +24,7 @@ func GetRegionFromKopsControlPlane(ctx context.Context, kcp *kcontrolplanev1alph
 		return nil, err
 	}
 
-	if err != nil {
```
Comment (csmartins) [inline pkg/kops/kops.go:27]: Don't we need to return the error if it happens?

File: internal/controller/ec2.aws/securitygroup_controller.go (hunk 1 above)
Comment (Rdpaula) [reply internal/controller/ec2.aws/securitygroup_controller.go:191]: It was correct, but we can simplify by omitting it.

Comment (csmartins) [reply internal/controller/ec2.aws/securitygroup_controller.go:191]: got it

File: pkg/kops/kops.go (hunk 3 above)
Comment (Rdpaula) [reply pkg/kops/kops.go:27]: We have already checked the error in [lines 23-25](https://github.com/topfreegames/kubernetes-crossplane-infrastructure-operator/blob/a9ed205487a84a4d65692bc598724d4806ee744b/pkg/kops/kops.go#L23-L25)

File: internal/controller/ec2.aws/securitygroup_controller.go (hunk 2 above)
Comment (Rdpaula) [reply internal/controller/ec2.aws/securitygroup_controller.go:978]: Firstly, I was only checking the first one, but I thought it was more reliable to get from the first one that can be found by the client. WDYT?

Comment (csmartins) [reply internal/controller/ec2.aws/securitygroup_controller.go:978]: Agreed! I misread the code and thought the nil check was out of the for. lgtm

**Review by csmartins — APPROVED** [review]

**Pull request merged**